src/domains/project/repository/project.repository.impl.ts

```diff
@@ -15,25 +24,84 @@ export class ProjectRepositoryImpl implements ProjectRepository {
     return project === null ? null : new ProjectDTO(project);
   }
 
-  async create(name: string, providerId: string, organizationId: string, image: string | null): Promise<ProjectDTO> {
-    const project: Project = await this.db.project.create({
-      data: {
-        name,
-        image,
+  /**
+   * Fetches a project by the ID the project has in the issue provider platform (ex. Linear).
+   *
+   * @param {string} providerId - ID of the external provider platform where the project is hosted.
+   * @returns {Promise<ProjectDTO | null>} - Promise resolving to a ProjectDTO object if found, otherwise null.
+   */
+  async getByProviderId(providerId: string): Promise<ProjectDTO | null> {
+    const project: Project | null = await this.db.project.findFirst({
+      where: {
         providerId,
-        organizationId,
       },
     });
-    return new ProjectDTO(project);
+
+    return project === null ? null : new ProjectDTO(project);
   }
 
-  async getByProviderId(providerId: string): Promise<ProjectDTO | null> {
-    const project: Project | null = await this.db.project.findFirst({
+  /**
+   * Fetches detailed information about a project, including associated users and roles.
+   *
+   * @param {string} projectId - ID of the project to retrieve details for.
+   * @param {string} userId - ID of the user requesting project details (used for access control).
+   * @returns {Promise<ProjectDetailsDTO | null>} - Promise resolving to a ProjectDetailsDTO object if found and user has access, otherwise null.
+   */
+  async getProjectDetails(projectId: string, userId: string): Promise<ProjectDetailsDTO | null> {
+    const details = await this.db.project.findFirst({
       where: {
-        providerId,
+        AND: [
+          {
+            id: projectId,
+          },
+          {
+            usersRoles: {
+              some: {
+                userId,
+              },
+            },
+          },
+        ],
+      },
+      include: {
+        usersRoles: {
+          select: {},
```

Comment:
Please correct the query, you cant add select and include both... I attach the error message returned when the endpoint was tested with swagger
![image](https://github.com/sirius-valley/tricker-backend/assets/104390122/9e1eae08-af76-4bb4-9d39-ce3ed3d4663f)
![image](https://github.com/sirius-valley/tricker-backend/assets/104390122/42c73d1e-6a6d-4a46-833e-c627dfafbd61)